Add test for rolling pool update with VM availability check on updated host

rushikeshjadhav opened the pull request:
Added `test_linstor_sr_pool_update` for rolling pool update with VM availability check.

It performs
- A rolling update and reboot of all hosts in a LINSTOR SR pool (starting with the master).
- Verifies VM can start and shutdown successfully on each host after update.
- Ensures SR remains usable throughout the process.

stormi commented on tests/storage/linstor/test_linstor_sr.py:
The tricky part here is ensuring that we do have packages to test the rolling pool update. That means starting with a pool which has an older version of the packages compared to the ones we want to test. That also likely means moving this kind of test to a specific job, or have an orchestrator test (or fixtures) that prepares a nested pool for this test.

See with @ydirson, as I think there are common challenges with the installation and upgrade tests.

rushikeshjadhav replied:
So should we limit this to `xcp-ng-linstor-testing` packages to keep it in linstor tests? Along side of main update tests, this can be specific for linstor related testing package tests.

Larger installation and upgrade tests can happen in different job.

```
# repoquery -a --repoid=xcp-ng-linstor-testing
linstor-common-0:1.29.2-1.el7_9.noarch
linstor-controller-0:1.29.2-1.el7_9.noarch
linstor-satellite-0:1.29.2-1.el7_9.noarch
```

```
# repoquery -a --repoid=xcp-ng-linstor
linstor-common-0:1.29.0-1.el7_9.noarch
linstor-controller-0:1.29.0-1.el7_9.noarch
linstor-satellite-0:1.29.0-1.el7_9.noarch
```

stormi commented on tests/storage/linstor/test_linstor_sr.py:
```diff
@@ -131,6 +131,32 @@ def test_linstor_missing(self, linstor_sr, host):
+        for h in hosts:
+            logging.info("Checking on host %s", h.hostname_or_ip)
+            if h.has_updates():
+                h.install_updates()
+                h.reboot(verify=True)
+                vm.start(on=h.uuid)
+                vm.wait_for_os_booted()
+                vm.shutdown(verify=True)
```
This doesn't look like the proper way to upgrade a pool with linstor. They don't support rolling upgrade, so you have to update all satellites and restart them, at some point. @Wescoeur has this documented somewhere.

rushikeshjadhav replied:
Yes, it's https://docs.xcp-ng.org/management/updates/#rolling-pool-update-rpu I have updated the code considering above.

rushikeshjadhav committed:
Define 'upgrade_test' marker for tests that validate package upgrades from testing repo

Signed-off-by: Rushikesh Jadhav <rushikesh7@gmail.com>

rushikeshjadhav committed:
Add upgrade_test marker detection and conditionally select testing repo in pool_with_linstor
This mechanism allows the test to start with our without testing repo packages

Signed-off-by: Rushikesh Jadhav <rushikesh7@gmail.com>

rushikeshjadhav committed:
Add enablerepo support to yum update and check-update in Host methods
Tests that want to test linstor upgrade can enablerepo during execution

Signed-off-by: Rushikesh Jadhav <rushikesh7@gmail.com>

rushikeshjadhav committed:
Added `test_linstor_sr_pool_update` for pool update with VM availability check.
- Uses `@pytest.mark.upgrade_test` to mark need of upgrade during the test
- Updates of all hosts in a LINSTOR SR pool (starting with the master)
- Reboots updated hosts
- Verifies VM can start and shutdown successfully on each host after update
- Ensures SR remains usable throughout the process

Signed-off-by: Rushikesh Jadhav <rushikesh7@gmail.com>

ydirson reviewed (changes requested):
In addition, please format the commit message [as per the standard converntions](https://www.baeldung.com/ops/git-commit-messages#bd-1-structure-for-good-commit-messages)
